### PR TITLE
fix(compiler-cli): allow relative rootDir

### PR DESCRIPTION
**Pull request by literalpie**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) N/A

## PR Type
What kind of change does this PR introduce?

- [X] Bugfix

## What is the current behavior?

Issue Number: #36290

## What is the new behavior?

If a relative rootDir is passed in, an error is no longer thrown

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

## Other information

Building off of #36291 by @Toxicable with test added and other recommended changes.

### CODE REVIEW

**Review by petebacondarwin — APPROVED** [review]
Great! Thanks @literalpie - I think this unit test is enough.
Can you modify the commit message? It doesn't need to mention the previous PR particularly, IMO. Perhaps:

```
fix(compiler-cli): resolve `rootDirs` to absolute paths

Ensure that `rootDirs` are absolute by resolving them against the
current working directory.

Fixes #36290
```

**Comment by literalpie** [comment]
@petebacondarwin done!

**Comment by jbedard** [comment]
Thanks for picking this up @literalpie!

**Comment by JoostK** [comment]
I once prepped an example of how this could be testing in an "integration-like" test in https://github.com/JoostK/angular/commit/c08c753df10d0666baa09ed12cf7de9d10c6387a, as without such a test this is easy to regress in. @literalpie could you have a look at adding a test like that?

File: packages/compiler-cli/test/ngtsc/env.ts [hunk 1]
```diff
@@ -38,7 +39,7 @@ export class NgtscTestEnvironment {
    * Set up a new testing environment.
    */
   static setup(files?: Folder, workingDir: AbsoluteFsPath = absoluteFrom('/')):
-      NgtscTestEnvironment {
+      NgtscTestEnvironment {``
```
Comment (JoostK) [inline packages/compiler-cli/test/ngtsc/env.ts:42]: This looks like a typo

**Review by JoostK — APPROVED** [review]

**Comment by JoostK** [comment]
@literalpie CI is not yet happy with the second commit; it wants it to have a short body.

**Comment by literalpie** [comment]
@JoostK i think it's better now. Thanks for your help and for your patience

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:367933533:BASE:367933538:1618204991275:23f46da7).

**Comment by JoostK** [comment]
@literalpie Thanks again for your contribution!!

**Pull request closed without merging**